### PR TITLE
FIX: Allow 1 column exog plots

### LINKED ISSUE #2364: bug or uncaught exception when plotting formula w/o constant
Thanks for the great tool! I've been learning how to use it the last few days. I've encountered a problem. I'm not sure if it's my fault, so apologies if it is!

Adapted from one of your examples:
```
from __future__ import print_function
from statsmodels.compat import lzip
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
from statsmodels.formula.api import ols

prestige = sm.datasets.get_rdataset("Duncan", "car", cache=True).data

print(prestige.head())

prestige_model = ols("prestige ~ education -1", data=prestige).fit()
#prestige_model = ols("prestige ~ income + education", data=prestige).fit()

print(prestige_model.summary())

fig = plt.figure(figsize=(12,8))
fig = sm.graphics.plot_regress_exog(prestige_model, "education", fig=fig)
```

The code pasted above gives the error pasted below. If I uncomment the line, the error goes away.

```
Traceback (most recent call last):
  File "test.py", line 19, in <module>
    fig = sm.graphics.plot_regress_exog(prestige_model, "education", fig=fig)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/graphics/regressionplots.py", line 211, in plot_regress_exog
    exog_others, obs_labels=False, ax=ax)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/graphics/regressionplots.py", line 339, in plot_partregress
    res_yaxis = OLS(endog, RHS).fit()
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/regression/linear_model.py", line 691, in __init__
    hasconst=hasconst, **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/regression/linear_model.py", line 586, in __init__
    weights=weights, hasconst=hasconst, **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/regression/linear_model.py", line 91, in __init__
    super(RegressionModel, self).__init__(endog, exog, **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/base/model.py", line 186, in __init__
    super(LikelihoodModel, self).__init__(endog, exog, **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/base/model.py", line 60, in __init__
    **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/base/model.py", line 84, in _handle_data
    data = handle_data(endog, exog, missing, hasconst, **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/base/data.py", line 566, in handle_data
    **kwargs)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/base/data.py", line 75, in __init__
    self._handle_constant(hasconst)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/statsmodels-0.6.1-py2.7-linux-x86_64.egg/statsmodels/base/data.py", line 135, in _handle_constant
    rank_orig = np_matrix_rank(self.exog)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 1500, in matrix_rank
    S = svd(M, compute_uv=False)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 1274, in svd
    _assertNonEmpty(a)
  File "/home/amosa/pkg/tools/python/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 170, in _assertNonEmpty
    raise LinAlgError("Arrays cannot be empty")
numpy.linalg.linalg.LinAlgError: Arrays cannot be empty
```

### PR DESCRIPTION
**Pull request by saketkc**
Fixes #2364

### CODE REVIEW

**Comment by josef-pkt** [comment]
@saketkc  Did you look at an example? I don't think that works correctly.

You could compare the plots of a model with one continuous variable with and without the constant added. I think that should produce the same plot, except for maybe the axis scaling or labeling.

**Comment by saketkc** [comment]
I did.
```
prestige_model = ols("prestige ~ education -1", data=prestige).fit()
fig = plt.figure(figsize=(12,8))
fig = sm.graphics.plot_regress_exog(prestige_model, "education", fig=fig)
```
With this change:

figure_2

```
prestige_model = ols("prestige ~ income + education", data=prestige).fit()
fig = plt.figure(figsize=(12,8))
fig = sm.graphics.plot_regress_exog(prestige_model, "education", fig=fig)
```
figure_1

**Comment by saketkc** [comment]
The partial regression doesn't make sense in this case, right?

**Comment by saketkc** [comment]
Wait, I think the partial regression plot is wrong.

**Comment by josef-pkt** [comment]
the model for the second graph should have education and constant (not additionally income which will change all results)

`ols("prestige ~ income", ...`    without the ` - 1` compared to the first model

partial regression plot still makes sense in this case, but it is redundant, it should be identical to one of the plots in right hand column (I don't remember which, I think it should be identical to the residual plot.)

**Comment by saketkc** [comment]
Yes, the partial regression plot is not correct. It should be identical to CCPR, as good as a scatter plot with single column exogs.

**Comment by josef-pkt** [comment]
starting at line 151

```
    # all arrays or pandas-like
    res_yaxis = OLS(endog, RHS).fit()
    res_xaxis = OLS(exog_i, RHS).fit()

    ax.plot(res_xaxis.resid, res_yaxis.resid, 'o', **kwargs)
    fitted_line = OLS(res_yaxis.resid, res_xaxis.resid).fit()
```

The first two OLS remove the effect of the other exog from endog and exog_i when we look at the residuals of those two regressions. 
The last two lines plot the residuals and fit a line to it. 

If the RHS is empty then we can skip the first two OLS and the original arrays should replace the `resid` in the last wo lines in the plot
use 
`endog` instead of `res_yaxis.resid`, and
`exog` instead of `res_xaxis.resid`

currently we use smoke tests for the plots, see the corresponding test modules to see where to add another smoke test for this case.

Thanks

**Comment by saketkc** [comment]
Will fix.
Thanks!

**Comment by saketkc** [comment]
@josef-pkt I could not think of a better way to do this. Let me know if it looks okay.

File: statsmodels/graphics/tests/test_regressionplots.py [hunk 1]
```diff
@@ -111,6 +111,18 @@ def setup(self):
         y = Series(y, name="outcome")
         res = sm.OLS(y, exog0).fit()
         self.res = res
+        data = DataFrame(exog0, columns=["const", "var1", "var2"])
+        data['y'] = y
+        self.data = data
+
+class TestPlotFormula(TestPlotPandas):
+    def test_one_column_exog(self):
```
Comment (josef-pkt) [inline statsmodels/graphics/tests/test_regressionplots.py]: I guess this needs to be decorated with a `skipif` when matplotlib is not available. 
Can you check the other test functions to see what we do here?

**Comment by josef-pkt** [comment]
Looks fine to me, reading through the changes. I might have time to try it out tomorrow.

**Comment by saketkc** [comment]
Fixed with lot of dirty commits. Let me know if it is okay, and I will rebase.

**Comment by josef-pkt** [comment]
You could squash it into one commit when TravisCI is green. with interactive rebase

File: statsmodels/graphics/tests/test_regressionplots.py [hunk 2]
```diff
@@ -27,15 +23,12 @@
 else:
     pdf = None
 
-def setup():
-    if not have_matplotlib:
-        raise nose.SkipTest('No tests here')
-
```
Comment (josef-pkt) [inline statsmodels/graphics/tests/test_regressionplots.py]: I didn't realize that the entire module is skipped.

**Comment by josef-pkt** [comment]
I didn't realize this was squashed and rebased.

Merging now
@saketkc  Thank you

**Pull request merged**